### PR TITLE
Handle the merge fields with tracking changes

### PR DESCRIPTION
**Pull request by fp52oxe**
TP Card: https://sycamoreinformatics.tpondemand.com/entity/19129-document-placeholders-may-break-when-a

### CODE REVIEW

File: lib/sablon/parser/mail_merge.rb [hunk 1]
```diff
@@ -108,9 +109,30 @@ def replace(content, env)
           end
 
           unless env.remove_fields_only
-            replace_field_display(pattern_node, content, env)
-            (@nodes - [pattern_node]).each(&:remove) unless env.keep_merge_fields
+            the_pattern_node = pattern_node(!inserting_word_ml)
+            replace_field_display(the_pattern_node, content, env)
+            if env.keep_merge_fields
```
Comment (rpfondeker) [inline lib/sablon/parser/mail_merge.rb:114]: which `env` is this referring to? @fp52oxe   Not the ssd app env?

File: lib/sablon/parser/mail_merge.rb [hunk 2]
```diff
@@ -198,15 +232,19 @@ def parse_fields(xml)
...
+      def build_complex_field(node, allow_recursion = true)
         possible_field_node = node.parent
         field_nodes = [possible_field_node]
         while possible_field_node && possible_field_node.search(".//w:fldChar[@w:fldCharType='end']").empty?
           possible_field_node = possible_field_node.next_element
           field_nodes << possible_field_node
         end
-        # skip instantiation if no end tag
-        ComplexField.new(field_nodes) if field_nodes.last
+        # skip instantiation if no end tag but try once again if it seems to be caused by tracking changes
```
Comment (rpfondeker) [inline lib/sablon/parser/mail_merge.rb:242]: 👍

**Review by rpfondeker — APPROVED** [review]
@fp52oxe Need to add `workspace.xml` to `.gitignore`.  One question then LGTM.  As you planned, please test this thoroughly on staging and let me know if you need my help with saving docs in Office 365.

File: lib/sablon/parser/mail_merge.rb (hunk 1 above)
Comment (fp52oxe) [reply lib/sablon/parser/mail_merge.rb:114]: @rpfondeker It's their internal collection of options

**Pull request merged**